### PR TITLE
Improve PluginManagerMock validation

### PR DESCRIPTION
**Pull request by Insprill**
# Description
- Add missing null checks
- Add missing argument checks
- Add missing plugin name validation

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
- Make sure that unit tests are added which test the relevant changes.
- Make sure that the changes follow the existing code format.

# For maintainer
When a PR is approved, the maintainer should label this PR with `release/*`.

- If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
- If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java [hunk 1]
```diff
@@ -69,6 +72,8 @@
 public class PluginManagerMock implements PluginManager
 {
 
+	private static final Pattern VALID_NAMES = Pattern.compile("^[A-Za-z0-9_.-]+$");
```
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java]: `VALID_PLUGIN_NAMES` may be more descriptive.

File: src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java [hunk 2]
```diff
@@ -361,13 +388,28 @@ public void registerLoadedPlugin(@NotNull Plugin plugin)
...
+		Preconditions.checkNotNull(description, "Description cannot be null");
+		Preconditions.checkNotNull(parameters, "Parameters cannot be null");
+
+		String name = description.getName();
+		if (name.equalsIgnoreCase("bukkit") || name.equalsIgnoreCase("minecraft") || name.equalsIgnoreCase("mojang"))
+		{
+			throw new RuntimeException("Restricted Name");
+		}
+
+		if (!VALID_NAMES.matcher(name).matches()) {
```
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java]: formatting
```suggestion
		if (!VALID_NAMES.matcher(name).matches())
		{
```

File: src/test/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMockTest.java [hunk 3]
```diff
@@ -371,4 +374,25 @@ public void event(BlockBreakEvent e) throws Exception
...
+
+		PluginDescriptionFile bukkit = new PluginDescriptionFile("bukkit", "1.0.0", TestPlugin.class.getName());
+		assertThrows(RuntimeException.class, () -> pluginManager.loadPlugin(TestPlugin.class, bukkit));
+
+		PluginDescriptionFile minecraft = new PluginDescriptionFile("minecraft", "1.0.0", TestPlugin.class.getName());
+		assertThrows(RuntimeException.class, () -> pluginManager.loadPlugin(TestPlugin.class, minecraft));
+
+		PluginDescriptionFile mojang = new PluginDescriptionFile("mojang", "1.0.0", TestPlugin.class.getName());
+		assertThrows(RuntimeException.class, () -> pluginManager.loadPlugin(TestPlugin.class, mojang));
+	}
```
Comment (TheBusyBiscuit) [inline src/test/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMockTest.java]: This should definitely be a parameterized test

**Review by TheBusyBiscuit — DISMISSED** [review]

File: src/test/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMockTest.java (hunk 3 above)
Comment (Insprill) [reply src/test/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMockTest.java]: Right, those exist 😅

**Review by thelooter — APPROVED** [review]
LGTM

**Pull request merged**